cpp/CMakeLists.txt

```diff
@@ -72,12 +72,8 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "10")
         message(WARNING "Clang <10 is not supported")
     endif()
-elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "10")
-        message(WARNING "GCC <10 is not supported")
-    endif()
-else() 
-    message(WARNING "Unsuported compiler, use Clang >10 or GCC >10")
+else()
+    message(WARNING "Unsuported compiler, use Clang >10")
```

Comment:
If we're going to update our supported compilers, we should test whether the code actually compiles with the clang >10 stdlib as well.